lock/src/main/java/org/infinispan/lock/api/impl/manager/EmbeddedClusteredLockManager.java

```diff
@@ -0,0 +1,152 @@
+   }
+
+   @Inject
+   public void injectDep(@ComponentName(KnownComponentNames.TIMEOUT_SCHEDULE_EXECUTOR) ScheduledExecutorService scheduledExecutorService,
+                         @ComponentName(KnownComponentNames.ASYNC_OPERATIONS_EXECUTOR) Executor executor) {
+      this.scheduledExecutorService = scheduledExecutorService;
+      this.executor = executor;
+   }
+
+   @Override
+   public boolean defineLock(String name) {
+      return defineLock(name, new ClusteredLockConfiguration());
+   }
+
+   @Override
+   public boolean defineLock(String name, ClusteredLockConfiguration configuration) {
+      // TODO: Configuration is not used because we don't support any other mode for now. For that : ISPN-8413
+      CacheHolder cacheHolder = extractCacheHolder(cacheHolderFuture);
+      cache = cacheHolder.getClusteredLockCache();
+      ClusteredLockKey key = new ClusteredLockKey(ByteString.fromString(name));
+      ClusteredLockValue clusteredLockValue = cache.putIfAbsent(key, new ClusteredLockValue());
+      locks.putIfAbsent(name, new ClusteredLockImpl(name, key, cache, this));
+      return clusteredLockValue == null;
+   }
+
+   @Override
+   public ClusteredLock get(String name) {
+      if (cache == null) {
+         cache = extractCacheHolder(cacheHolderFuture).getClusteredLockCache();
+      }
+
+      return locks.computeIfAbsent(name, k -> {
+         ClusteredLockKey key = new ClusteredLockKey(ByteString.fromString(k));
+         if (!cache.containsKey(key)) {
+            throw new ClusteredLockException(String.format("Lock does %s not exist", name));
+         }
+         return new ClusteredLockImpl(k, key, cache, this);
+      });
+   }
+
+   @Override
+   public ClusteredLockConfiguration getConfiguration(String name) {
+      CacheHolder cacheHolder = extractCacheHolder(cacheHolderFuture);
+      cache = cacheHolder.getClusteredLockCache();
+      if (cache.containsKey(new ClusteredLockKey(ByteString.fromString(name)))) {
+         return new ClusteredLockConfiguration();
+      }
+      throw new ClusteredLockException(String.format("Lock does %s not exist", name));
+   }
+
+   @Override
+   public boolean isDefined(String name) {
+      if (cache == null) {
+         cache = extractCacheHolder(cacheHolderFuture).getClusteredLockCache();
+      }
+      return cache.containsKey(new ClusteredLockKey(ByteString.fromString(name)));
+   }
+
+   @Override
+   public CompletableFuture<Boolean> remove(String name) {
+      CacheHolder cacheHolder = extractCacheHolder(cacheHolderFuture);
+      AdvancedCache<ClusteredLockKey, ClusteredLockValue> clusteredLockCache = cacheHolder.getClusteredLockCache();
+      return clusteredLockCache.removeAsync(new ClusteredLockKey(ByteString.fromString(name))).thenApply(value -> {
+         locks.remove(name);
+         return value != null;
+      });
+   }
+
+   @Override
+   public CompletableFuture<Boolean> forceRelease(String name) {
+      return CompletableFuture.supplyAsync(() -> {
```

Comment:
You could use functional API which is async by nature instead of executing this in FJP:
```
rwmap.eval(new ClusteredLockKey(ByteString.fromString(name)), view -> if (view.find().isPresent()) {
   boolean retval = view.find().getState() == ClusteredLockState.RELEASED;
   view.set(new ClusteredLockValue());
   return retval;
} else {
   return false;
});
```